addons/l10n_it_edi_ndd/models/account_move.py

```diff
@@ -58,7 +58,11 @@ def _compute_l10n_it_payment_method(self):
     def _compute_l10n_it_document_type(self):
         document_type = self.env['l10n_it.document.type'].search([]).grouped('code')
         for move in self:
-            if move.country_code != 'IT' or move.l10n_it_document_type or move.state != 'posted':
+            if move.country_code != 'IT':
+                continue
+            if move.debit_origin_id:
+                move.l10n_it_document_type = document_type.get('TD05')
+            if move.l10n_it_document_type or move.state != 'posted':
```

Comment:
this might required a comment on why you have to set the document type before doing the continue, i was mind fucked by the diff at first ahah
Also maybe a test would be nice for the change you made :smile: